File: Barberly/app/backend/Models/Timeslot.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models;

public class Timeslot
{
    [Key]
    public Guid timeslotId { get; set; } = Guid.NewGuid();
    public DateOnly date { get; set; }
    public TimeOnly startTime { get; set; }
    public int duration { get; set; }

    public Salon? salon { get; set; }
    public Barber? barber { get; set; }
}

File: Barberly/app/frontend/src/models/Timeslot.ts
import type { Barber } from "./Barber";
import type { Salon } from "./Salon";

export type Timeslot = {
  timeslotId: string;
  date: string;
  startTime: string;
  duration: number;
  isBooked: boolean;
  salon?: Salon;
  barber?: Barber;
};

export interface CreateTimeslotDto {
  barberId: string;
  salonId: string | null;
  date: string;
  startTime: string;
  duration: number;
  isBooked?: boolean;
}

export interface UpdateTimeslotDto {
  date: string;
  startTime: string;
  duration: number;
  salonId: string;
  barberId: string;
  isBooked: boolean;
}
